mmv1/third_party/terraform/services/bigquery/resource_bigquery_table.go.tmpl

```diff
@@ -96,6 +96,14 @@ func jsonCompareWithMapKeyOverride(key string, a, b interface{}, compareMapKeyVa
 			unionOfKeys[subKey] = true
 		}
 
+		{{- if ne $.TargetVersionName "ga" }}
+		// Disregard "type" and "fields" if "foreignTypeDefinition" is present since they may have been modified by the server.
+		if _, ok := unionOfKeys["foreignTypeDefinition"]; ok {
+			delete(unionOfKeys, "type")
+			delete(unionOfKeys, "fields")
```

Comment:
Question - Would users be able to change type and fields if foreignTypeDefinition is present?

Reply:
User-supplied values will be ignored. Service will re-populate type and fields based on the given foreignTypeDefinition value.